File: xkcd/src/main/java/xyz/jienan/xkcd/comics/contract/ImageDetailPageContract.java
package xyz.jienan.xkcd.comics.contract;

import android.graphics.Bitmap;

import xyz.jienan.xkcd.base.BasePresenter;
import xyz.jienan.xkcd.base.BaseView;
import xyz.jienan.xkcd.model.XkcdPic;

public interface ImageDetailPageContract {

    interface View extends BaseView<Presenter> {

        void setLoading(boolean isLoading);

        void renderPic(String targetImgUrl);

        void renderTitle(XkcdPic xkcdPic);

        void renderSeekBar(int d);

        void renderFrame(Bitmap bitmap);

        void changeGifSeekBarProgress(int progress);

        void showGifPlaySpeed(int speed);
    }

    interface Presenter extends BasePresenter {

        void requestImage(int index);

        void parseGifData(byte[] data);

        void parseFrame(int progress);

        void adjustGifSpeed(int increaseByOne);

        void adjustGifFrame(boolean isForward);

        void setEcoMode(boolean isEcoMode);
    }
}

File: xkcd/src/main/java/xyz/jienan/xkcd/comics/contract/ImageDetailPageContract.kt
package xyz.jienan.xkcd.comics.contract

import android.graphics.Bitmap

import xyz.jienan.xkcd.base.BasePresenter
import xyz.jienan.xkcd.base.BaseView
import xyz.jienan.xkcd.model.XkcdPic

interface ImageDetailPageContract {

    interface View : BaseView<Presenter> {

        fun setLoading(isLoading: Boolean)

        fun renderPic(url: String)

        fun renderTitle(xkcdPic: XkcdPic)

        fun renderSeekBar(duration: Int)

        fun renderFrame(bitmap: Bitmap)

        fun changeGifSeekBarProgress(progress: Int)

        fun showGifPlaySpeed(speed: Int)
    }

    interface Presenter : BasePresenter {

        fun requestImage(index: Int)

        fun parseGifData(data: ByteArray?)

        fun parseFrame(progress: Int)

        fun adjustGifSpeed(increaseByOne: Int)

        fun adjustGifFrame(isForward: Boolean)

        var isEcoMode: Boolean
    }
}
